sdk/nodejs/stackReference.ts

```diff
@@ -91,9 +91,15 @@ export class StackReference extends CustomResource {
      * are secrets.
      *
      * @param name The name of the stack output to fetch.
+     * @deprecated
      */
     public getOutputSync(name: string): any {
         const out = this.getOutput(name);
+
+        // Have to do an explicit console.log here as the call to utils.promiseResult may hang
+        // node, and that may prevent our normal logging calls from making it back to the user.
+        console.log(`"getOutputSync" is deprecated and may cause the program to hang. Please use "getOutput" instead.`);
```

Comment:
this is unfortunate, and is due to how we use a normal resource+outputs to model this relationship.  We could consider making it possible to get stackreference outputs synchronously if we handled it like we do 'invokes'.  i.e. by actually making a synchronous call to the engine (using our file-handles) to get this information.

Reply:
I'm particularly worried about this: IIRC we've strongly pushed users towards `getOutputSync`, and I don't know that simply deprecating this API is acceptable. @lukehoban thoughts?

Reply:
Yeah - I'm nervous about this too.  I think we will have to provide some sort of warning here - as these will nearly always hang the program on Node 12 and 13.  But the fact that we have no real answer for what to do instead yet is worrisome.

Net though, it's not clear it's better to do nothing than it is to warn on these.  

@pgavlin Any other option you think we should consider here?

Reply:
> and I don't know that simply deprecating this API is acceptable. @lukehoban thoughts?

If deprecating is not acceptable, can we move to a model where getOutputSync operates similarly to an "invoke".  i.e. we actually make a synchronous call over the wire (using our safe mechanism to do so) to get the json value back?

Reply:
We now support getting outputs synchronous in a safe manner for common cases.  we only warn in the case we can't do this.